filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/stream/config/AkkaRunnableGraph.java

```diff
@@ -177,41 +191,92 @@ FilteredResourceSourceFactory filteredResourceSourceFactory(final PartitionedOff
-                            );
-                        }
-                        return tokenMarkerRequestCommittable;
-                    })
+                    .alsoTo(
+                            Flow.<Tuple4<String, Optional<StreamMarker>, Optional<FilteredResourceRequest>, CommittableMessage<String, FilteredResourceRequest>>>create()
+                                    .filter(tokenMarkerRequestCommittable -> tokenMarkerRequestCommittable.t3()
+                                            // If there are resources, set the observedResource boolean to true
+                                            .map((FilteredResourceRequest request) -> {
+                                                observedResource.set(true);
+                                                return false;
+                                            })
+                                            // If we haven't seen any resources, but recieve the end of stream message, then audit it
+                                            .orElse(tokenMarkerRequestCommittable.t2()
+                                                    .filter(StreamMarker.END::equals)
+                                                    .filter(endMarker -> !observedResource.get())
+                                                    .isPresent()))
+
+                                    .map((Tuple4<String, Optional<StreamMarker>, Optional<FilteredResourceRequest>, CommittableMessage<String, FilteredResourceRequest>> tokenMarkerRequestCommittable) -> {
+                                        // Extract the headers
+                                        Headers headers = tokenMarkerRequestCommittable.t4().record().headers();
+
+                                        // Build the error message
+                                        AuditErrorMessage auditErrorMessage = AuditErrorMessage.Builder.create()
+                                                .withUserId(UNKNOWN)
+                                                .withResourceId(UNKNOWN)
+                                                .withContext(new Context().purpose(UNKNOWN))
+                                                .withAttributes(Collections.emptyMap())
+                                                .withError(new NoResourcesObservedException("No Resources were observed for token: " + tokenMarkerRequestCommittable.t1()));
+
+                                        LOGGER.debug("NoResourcesObservedException thrown for token {}, on partition {} and topic {}", tokenMarkerRequestCommittable.t1(), partition, errorTopic.getName());
+                                        // Create the ProducerRecord, on the error topic, on the right partition, with the audit error message
+                                        return new ProducerRecord<>(errorTopic.getName(), partition, (String) null, auditErrorMessage, headers);
```

Comment:
Why not just throw the the `.getError()` exception?
```suggestion
                                        throw x.value().getError();
```